app/actors/curation_concerns/file_actor.rb

```diff
@@ -0,0 +1,92 @@
+module CurationConcerns
+  # actions for a file identified by file_set and relation (maps to use predicate)
+  class FileActor
+    attr_reader :file_set, :relation, :user
+
+    # @param [FileSet] file_set the parent FileSet
+    # @param [String] relation the type/use for the file.
+    # @param [User] user the user to record as the Agent acting upon the file
+    def initialize(file_set, relation, user)
+      @file_set = file_set
+      @relation = relation
+      @user = user
+    end
+
+    # Puts the uploaded content into a staging directory. Then kicks off a
+    # job to characterize and create derivatives with this on disk variant.
+    # Simultaneously moving a preservation copy to the repostiory.
+    # TODO: create a job to monitor this directory and prune old files that
+    # have made it to the repo
+    # @param [File, ActionDigest::HTTP::UploadedFile, Tempfile] the file to save in the repository
+    def ingest_file(file)
+      working_file = copy_file_to_working_directory(file, file_set.id)
+      IngestFileJob.perform_later(file_set, working_file, mime_type(file), user.user_key, relation)
+      make_derivative(file_set, working_file)
+      true
+    end
+
```

Comment:
Oh, I thought it would be easier.  But I see what you're saying about having to override several methods in FileSetActor.  So maybe this is the best way for now.

I'll make a PR for CurationConcerns to at least have a FileSetActor#file_actor method, which will make it somewhat easier to override.  No need to wait for that, but we could use that assuming it's merged quickly.